src/lib.rs

```diff
@@ -20,13 +20,16 @@ pub enum WeightUnit {
 }
 
 fn setup_store<T>(store: &mut Store<T>, weight_unit: WeightUnit) {
+    let fuel_max: u64 = store.get_fuel().unwrap();
     store.set_epoch_deadline(1);
     store.epoch_deadline_callback(move |context| {
         if let Some(ticker) = unsafe { TICKER.as_ref() } {
             let mut backtraces = BACKTRACES.lock().unwrap();
+            let current_fuel = context.get_fuel().unwrap();
+            let fuel_consumption = fuel_max.saturating_sub(current_fuel);
```

Comment:
also this is partially wrong, as we want to know how much fuel was consumed in between each tick, not since the start. so we should also keep track of the last seen fuel and change at each step. I would recommend using an AtomicU64 for this.

Reply:
Hm- but the prior method used here `fuel_consumed` was documented like this:

>Returns the amount of fuel consumed by this context's store execution so far

What you're suggesting sounds like a change in behaviour from the prior art

Reply:
My understanding of the code was that the subsequent line `weight - last_weight` was what ensured that each backtrace only included it's own weight, not the measurement itself

Reply:
oh ok, maybe the code was bad then, sorry! but it feels like it would be more correct now to just keep track in between steps

Reply:
oh wait sorry im rusty on my own code 🤦 you're right this is ok